modules/gapi/test/serialization/gsrlzimgproc.cpp

```diff
@@ -0,0 +1,450 @@
+    {
+        cv::Mat rgb;
+        cv::cvtColorTwoPlane(inY, inUV, rgb, cv::COLOR_YUV2RGB_NV12);
+        toPlanar(rgb, out);
+    }
+};
+
+G_TYPED_KERNEL(GYUV2Gray, <cv::GMat(cv::GMat)>, "yuvtogray") {
+    static cv::GMatDesc outMeta(cv::GMatDesc in) {
+        GAPI_Assert(in.depth  == CV_8U);
+        GAPI_Assert(in.planar == false);
+        GAPI_Assert(in.size.width  % 2 == 0);
+        GAPI_Assert(in.size.height % 3 == 0);
+
+        /* YUV format for this kernel:
+         * Y Y Y Y Y Y Y Y
+         * Y Y Y Y Y Y Y Y
+         * Y Y Y Y Y Y Y Y
+         * Y Y Y Y Y Y Y Y
+         * U V U V U V U V
+         * U V U V U V U V
+         */
+
+        return {CV_8U, 1, cv::Size{in.size.width, in.size.height - (in.size.height / 3)}, false};
+    }
+};
+
+GAPI_SRLZ_KERNEL(GSRLZYUV2Gray, GYUV2Gray)
+{
+    static void run(const cv::Mat& in, cv::Mat& out)
+    {
+        cv::cvtColor(in, out, cv::COLOR_YUV2GRAY_NV12);
+    }
+};
+
+G_TYPED_KERNEL(GConcatYUVPlanes, <cv::GMat(cv::GMat, cv::GMat)>, "concatyuvplanes") {
+    static cv::GMatDesc outMeta(cv::GMatDesc y, cv::GMatDesc uv) {
+        return {CV_8U, 1, cv::Size{y.size.width, y.size.height + uv.size.height}, false};
+    }
+};
+
+GAPI_SRLZ_KERNEL(GSRLZConcatYUVPlanes, GConcatYUVPlanes)
+{
+    static void run(const cv::Mat& in_y, const cv::Mat& in_uv, cv::Mat& out)
+    {
+        cv::Mat uv_planar(in_uv.rows, in_uv.cols * 2, CV_8UC1, in_uv.data);
+        cv::vconcat(in_y, uv_planar, out);
+    }
+};
+
+GAPI_COMPOUND_KERNEL(GSRLZNV12toGray, cv::gapi::imgproc::GNV12toGray)
+{
+    static cv::GMat expand(cv::GMat y, cv::GMat uv)
+    {
+        return GYUV2Gray::on(GConcatYUVPlanes::on(y, uv));
+    }
+};
+
+GAPI_SRLZ_KERNEL(GSRLZNV12toBGRp, cv::gapi::imgproc::GNV12toBGRp)
+{
+    static void run(const cv::Mat& inY, const cv::Mat& inUV, cv::Mat& out)
+    {
+        cv::Mat rgb;
+        cv::cvtColorTwoPlane(inY, inUV, rgb, cv::COLOR_YUV2BGR_NV12);
+        toPlanar(rgb, out);
+    }
+};
+
+cv::gapi::GKernelPackage cv::gapi::imgproc::serialization::kernels()
+{
+    static auto pkg = cv::gapi::kernels
+        < GSRLZFilter2D
+        , GSRLZSepFilter
+        , GSRLZBoxFilter
+        , GSRLZBlur
+        , GSRLZGaussBlur
+        , GSRLZMedianBlur
+        , GSRLZErode
+        , GSRLZDilate
+        , GSRLZSobel
+        , GSRLZSobelXY
+        , GSRLZCanny
+        , GSRLZGoodFeatures
+        , GSRLZEqualizeHist
+        , GSRLZRGB2YUV
+        , GSRLZYUV2RGB
+        , GSRLZNV12toRGB
+        , GSRLZNV12toBGR
+        , GSRLZRGB2Lab
+        , GSRLZBGR2LUV
+        , GSRLZBGR2YUV
+        , GSRLZYUV2BGR
+        , GSRLZLUV2BGR
+        , GSRLZBGR2Gray
+        , GSRLZRGB2Gray
+        , GSRLZRGB2GrayCustom
+        , GSRLZBayerGR2RGB
+        , GSRLZRGB2HSV
+        , GSRLZRGB2YUV422
+        , GSRLZYUV2Gray
+        , GSRLZNV12toRGBp
+        , GSRLZNV12toBGRp
+        , GSRLZNV12toGray
+        , GSRLZConcatYUVPlanes
+        >();
+    return pkg;
+}
```

Comment:
Please remove all the kernels which are not required for your test pipelines.

Reply:
Done